.github/CONTRIBUTING.md

```diff
@@ -69,102 +67,142 @@ There are few ways you can play with the build:
 
-For environment stability, Web Chat uses Docker for hosting the test environment.
+### Running tests manually
 
--  Docker with [Windows Subsystem for Linux 2 (a.k.a. WSL2)](https://aka.ms/wsl2)
-   -  To start, run `npm run start:docker`
-   -  To stop/cleanup, run `npm run stop:docker`
--  Other Docker setup
-   -  To start, run `docker-compose up --build`
-   -  To stop/cleanup, run `docker-compose down`
+Download [ChromeDriver](https://sites.google.com/a/chromium.org/chromedriver/downloads) and extract to the project root.
 
-> On Windows, set environment variable `COMPOSE_CONVERT_WINDOWS_PATHS=1`.
+Run `npm run browser`. It will open a new browser window to http://localhost:5001/**tests**/html/. Then, navigate to the test file.
 
-### Running the test suite
+When tests have completed successfully, it should show a green check.
 
-> Testing pages under `__tests__/html/` are served under http://localhost:5001/ (locally) and http://localhost:5081/ (from Docker). All Jest tests will run against Docker version of the page. To update the page, restart Docker.
+![Transcript with a green check showing test succeeded](https://github.com/microsoft/BotFramework-WebChat/raw/main/docs/media/running-test.png)
 
-There are 2 ways to run the test suite, either one-off or continuously:
+> If the test pages take any screenshots, run them in an automated manner to save the screenshots.
 
-> To run Web Driver test, locally, download [ChromeDriver](https://sites.google.com/a/chromium.org/chromedriver/downloads) and extract to project root. Then set environment variable `WEBCHAT_TEST_ENV=chrome-local` before running Jest.
+### Running tests automatically
 
--  For one-off testing, `node_modules/.bin/jest`
--  For continuous testing, `npm test`
-   -  To speed up continuous testing, you can install [watchman](https://facebook.github.io/watchman/)
+For test environment convergence and stability, Web Chat uses Docker for hosting the test environment. Please install the following components:
 
-If your development box has less than 4 cores, you will need to reduce the number of simultaneous test agents:
+-  On Windows, install [Windows Subsystem for Linux 2 (a.k.a. WSL 2)](https://aka.ms/wsl2)
+-  [Docker Desktop](https://www.docker.com/get-started)
+-  (Optional) Install [watchman](https://facebook.github.io/watchman/) to improve Jest performance
 
--  For one-off testing: run `node_modules/.bin/jest --maxWorkers 1`
--  For continuous testing: run `npm test -- --maxWorkers 1`
+Run `npm run docker`. It will start a Docker Compose with Selenium Grid and 4 nodes of Chrome.
 
-Our CI pipeline run tests with 4 agents simultaneously. If new tests are added, please make sure they can run simultaneously.
+Then, run `npm test` to start Jest. When Jest run the test pages, it will take screenshots and save it under [`/__tests__/__image_snapshots__/html`](https://github.com/microsoft/BotFramework-WebChat/tree/main/__tests__/__image_snapshots__/html).
```

Comment:
```suggestion
Run `npm test` to start Jest. When Jest runs the test pages, it will take screenshots of new tests and save it under [`/__tests__/__image_snapshots__/html`](https://github.com/microsoft/BotFramework-WebChat/tree/main/__tests__/__image_snapshots__/html).
```